src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -175,6 +179,107 @@ public void testDBSchema() throws SQLException {
             assertTrue(!StringUtils.isEmpty(rs.getString(1)), form.format(msgArgs));
         }
     }
+    
+    /**
+     * Tests that the catalog parameter containing - is escaped by {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogName() throws SQLException {
+        UUID id = UUID.randomUUID();
+        String testCatalog = "dash-catalog" + id;
+        String testSchema = "some-schema" + id;
+        Statement stmt = connection.createStatement();
```

Comment:
You can move this into try with resources.

Reply:
Cannot resolve variables declared in try-with-resources in a finally block, and a finally block is used to drop the created db.